homeassistant/components/zha/number.py

```diff
@@ -553,6 +559,16 @@ def __init__(
             self._attr_native_min_value: float = self._channel.min_mireds
             self._attr_native_max_value: float = self._channel.max_mireds
```

Comment:
That was my first approach as well but didn't work because some component (I believe its [number/__init.py__#L99](https://github.com/home-assistant/core/pull/81684/files/c26f82ce424ad4dd968cb1cbb91cac8fc39124dc#diff-3f781bf9011e8d2a51366e5690c2dc4e372cd6845e3eb8d01dd3167ecef5a4eeR99)) does a range check unable to handle inversed ranges and fails with an error message.

So this requires a number integration change anyways. Therefore I opted to go down the reverse mapping support route and keep min and max values accordingly.